### PR TITLE
core.evaluate : evaluate(False) works for numeric operators (Fix for : #11614)

### PR DESCRIPTION
**Pull request by ArifAhmed1995**
Ref : #11614 

Certain fixes have been done to `sympify.py` and `numbers.py` to ensure that if numerical operations are executed in a `with evaluate(False):` block the unevaluated numerical expression is returned.

Example : 

Previously : 

```
>>> with evaluate(False):
...     print(S(1) + 4)
... 
5
```

With the fix :

```
>>> with evaluate(False):
...     print(S(1) + 4)
... 
1 + 4
```

Unit tests have also been added.

### CODE REVIEW

File: sympy/core/tests/test_evaluate.py [hunk 1]
```diff
@@ -15,7 +16,27 @@ def test_add():
 
     assert isinstance(x + x, Mul)
 
+    with evaluate(False):
+        assert S(1) + 1 == Add(1, 1)
```
Comment (asmeurer) [inline sympy/core/tests/test_evaluate.py]: Can you also add tests for the right operators (like `1 + S(1)`)?

**Comment by asmeurer** [comment]
Other than the one comment this looks good to me.

File: sympy/core/tests/test_evaluate.py (hunk 1 above)
Comment (ArifAhmed1995) [inline sympy/core/tests/test_evaluate.py]: Done.

**Comment by asmeurer** [comment]
I don't know why GitHub is showing some commits from master in here (the commit https://github.com/sympy/sympy/commit/d12dd1a2c6b1f066e8d798fc06a162ce7ed4b3e2 is the exact commit from master). I guess it's a GitHub bug. 

At any rate, this looks good to me.

**Pull request merged**